app/src/routes/api/v1/gfw.router.ts

```diff
@@ -0,0 +1,248 @@
+import { Context, Next } from 'koa';
+import Router from 'koa-router';
+import {
+    RWAPIMicroservice,
+    RequestToMicroserviceOptions,
+} from 'rw-api-microservice-node';
+import type request from 'request';
+
+import FieldSerializer from 'serializers/field.serializer';
+import logger from 'logger';
+import DatasetMiddleware from 'middleware/dataset.middleware';
+import GfwService from 'services/gfw.service';
+import ErrorSerializer from 'serializers/errorSerializer';
+
+const API_VERSION: string = 'v1';
+
+const router: Router = new Router({
+    prefix: `/api/${API_VERSION}/gfw`,
+});
+
+class GfwRouter {
+    static getCloneUrl(url: string, idDataset: string): Record<string, any> {
+        return {
+            http_method: 'POST',
+            url: `/dataset/${idDataset}/clone`,
+            body: {
+                dataset: {
+                    datasetUrl: url
+                        .replace('/gfw', '')
+                        .replace(`/api/${API_VERSION}`, ''),
+                    application: ['your', 'apps'],
+                },
+            },
+        };
+    }
+
+    static async registerDataset(ctx: Context): Promise<void> {
+        logger.info('Registering dataset with data', ctx.request.body);
+        try {
+            await GfwService.getFields(ctx.request.body.connector.connectorUrl);
+            await RWAPIMicroservice.requestToMicroservice({
+                method: 'PATCH',
+                uri: `/${API_VERSION}/dataset/${ctx.request.body.connector.id}`,
+                body: {
+                    dataset: {
+                        status: 1,
+                    },
+                },
+                json: true,
+            });
+        } catch (e) {
+            await RWAPIMicroservice.requestToMicroservice({
```

Comment:
what happens if this call to an external MS fails?

Reply:
Now catching that and raising proper error for that scenario